### PR TITLE
New JGroups plugin for 3.6.6.Final, features:

### PR DESCRIPTION
**Pull request by belaban**
- Various optimizations (RequestOptions are created up-front, not on every request)
- Mimics the way Infinispan does reads and writes more closely
  - Details in the javadoc
  - New attribute primary_replicates_puts (default: true) which mimicks the
    way Infinispan does writes
- GH issue: https://github.com/radargun/radargun/issues/295
- Changed defaults for oob (true), flowControl (true) and bundle (true)
  - Performance optimization
- Added (manual) test

### CODE REVIEW

File: plugins/jgroups36/src/main/java/org/radargun/service/JGroups36Service.java [hunk 1]
```diff
@@ -0,0 +1,494 @@
...
+        return ch != null && ch.isConnected();
+    }
+
+    public Object getFromRemote(Object key) {
+        assert key !=null;
+        return lastValue;
+    }
+
+    public boolean containsKeyFromRemote(Object key) {
+        assert this != null && key != null;
```
Comment (rvansa) [inline plugins/jgroups36/src/main/java/org/radargun/service/JGroups36Service.java]: `assert this != null`? I think this is Java, not C++

**Comment by rvansa** [comment]
Do you think it would be possible to default configurations provided in JGroups JAR, instead of adding those two files? Many plugins do the same, but the less files the better.

**Comment by rvansa** [comment]
Functionally it seems okay to me.

I would ignore some coding style oddities such as missing space after `if`, `for` or around operators, but please don't use underscores in variable names. And please remove those commented logs to System.out

We should add some basic coding style guidelines on wiki, and hopefully @mcimbora eventually writes a github hook to unify the formatting when issuing PR.

**Comment by belaban** [comment]
OK, try now.

**Comment by rvansa** [comment]
Please Ctrl + F for `exclude_rank`, `backup_owners`, `return_null_on_self_inclusion`, `skip_self`, `primary_rank`, `received_non_null_rsp` and `// System.out`

**Comment by rvansa** [comment]
I've amended the commit with coding-style fixes with few more and integrated this into master branch. Thanks! And looking forward to rewriting this to message-oriented traits.

**Comment by belaban** [comment]
ok, tkx

**Pull request closed without merging**